### ISSUE TITLE
Polyhedral mesh executions fail when no solution is present

### ISSUE THREAD

**Issue by mlohry** [Alpine-DAV/ascent#1260]
The following example creates a mesh of a single tetrahedron using the general polyhedral indexing format. It should then render a mesh image and extract an hdf5 file. What happens instead is:

```
[Error] Ascent::execute
file: conduit_node.cpp
line: 14084
message:
Cannot access non-existent child "fields" from Node([0])
```

it is looking for solution fields even though none are requested in the actions.

```
  std::vector<double> xvert = {0.0, 1.0, 0.0, 0.0};
  std::vector<double> yvert = {0.0, 0.0, 0.0, 1.0};
  std::vector<double> zvert = {1.0, 0.0, 0.0, 0.0};
  conduit::Node       mesh;
  mesh["coordsets/coords/type"] = "explicit";
  mesh["coordsets/coords/values/x"].set_external(xvert);
  mesh["coordsets/coords/values/y"].set_external(yvert);
  mesh["coordsets/coords/values/z"].set_external(zvert);

  std::vector<std::int32_t> poly_face_conn = {
      0,1,3,  // face 0
      1,2,3,  // face 1
      2,0,3,  // face 2
      0,2,1   // face 3
  };
  std::vector<std::int32_t> poly_face_sizes   = {3, 3, 3, 3};
  std::vector<std::int32_t> poly_face_offsets = {0, 3, 6, 9};
  std::vector<std::int32_t> poly_cell_conn    = {0, 1, 2, 3};
  std::vector<std::int32_t> poly_cell_sizes   = {4};
  std::vector<std::int32_t> poly_cell_offsets = {0};

  mesh["topologies/topo/type"]              = "unstructured";
  mesh["topologies/topo/elements/shape"]    = "polyhedral";
  mesh["topologies/topo/subelements/shape"] = "polygonal";
  mesh["topologies/topo/coordset"]          = "coords";
  mesh["topologies/topo/elements/connectivity"].set_external(poly_cell_conn);
  mesh["topologies/topo/elements/sizes"].set_external(poly_cell_sizes);
  mesh["topologies/topo/elements/offsets"].set_external(poly_cell_offsets);
  mesh["topologies/topo/subelements/connectivity"].set_external(poly_face_conn);
  mesh["topologies/topo/subelements/sizes"].set_external(poly_face_sizes);
  mesh["topologies/topo/subelements/offsets"].set_external(poly_face_offsets);

  ascent::Ascent ascent_viewer;
  conduit::Node  ascent_opts;
  ascent_viewer.open(ascent_opts);
  ascent_viewer.publish(mesh);

  conduit::Node actions;

  // declare an action to create a .png render
  conduit::Node& add_actions = actions.append();
  add_actions["action"]      = "add_scenes";
  conduit::Node& scenes      = add_actions["scenes"];
  scenes["s1/plots/p1/type"] = "mesh";
  scenes["s1/image_prefix"]  = "single_tet_poly_mesh_render";

  // declare an action to create a mesh extraction to hdf5
  conduit::Node& add_extracts    = actions.append();
  add_extracts["action"]         = "add_extracts";
  conduit::Node& extracts        = add_extracts["extracts"];
  extracts["e0/type"]            = "relay";
  extracts["e0/params/path"]     = "single_tet_poly_mesh_extract";
  extracts["e0/params/protocol"] = "blueprint/mesh/hdf5";

  ascent_viewer.execute(actions);
  ascent_viewer.close();
```

If I insert some dummy data, then the mesh render succeeds,

```
  std::vector<double> ones(xvert.size(), 1.0);
  mesh["fields/dummy/association"] = "vertex";
  mesh["fields/dummy/topology"] = "topo";
  mesh["fields/dummy/values"].set_external(ones);
```

 The same test code but using a tetrahedral type indexing (not the general polyhedral) like the following works as expected and correctly renders the mesh.

```
  std::vector<double> xvert = {0.0, 1.0, 0.0, 0.0};
  std::vector<double> yvert = {0.0, 0.0, 0.0, 1.0};
  std::vector<double> zvert = {1.0, 0.0, 0.0, 0.0};
  std::vector<int>    conn  = {0, 1, 2, 3};
  conduit::Node       mesh;
  mesh["coordsets/coords/type"] = "explicit";
  mesh["coordsets/coords/values/x"].set_external(xvert);
  mesh["coordsets/coords/values/y"].set_external(yvert);
  mesh["coordsets/coords/values/z"].set_external(zvert);
  mesh["topologies/topo/type"]           = "unstructured";
  mesh["topologies/topo/elements/shape"] = "tet";
  mesh["topologies/topo/coordset"]       = "coords";
  mesh["topologies/topo/elements/connectivity"].set_external(conn);
```

**Comment by cyrush** [comment]
@mlohry  thanks for the report, I found the issue. 

The current logic that generates the sides mesh for rendering assumes fields are present.

https://github.com/Alpine-DAV/ascent/blob/08b8c0cc2825a0f9be70df41cfcd4a1a70eb2497/src/libs/ascent/runtimes/ascent_transmogrifier.cpp#L138

I will add logic to check for fields and if they aren't there use a different generate call.

Related:
https://github.com/LLNL/conduit/issues/1265

**Comment by cyrush** [comment]
resolved in #1262
